### PR TITLE
Refact invite collection handler

### PR DESCRIPTION
**Pull request by Luiz-FS**
**Feature/Bug description:** The invite_collection_handler handler was very confusing because it was responsible for handling many types of invitation (INSTITUTION_CHILDREN, INSTITUTION_PARENT, USER, USER_ADM), which made its logic very confusing and difficult to understand.

**Solution:** Create two new handlers to handle invitations in separate locations. The handlers created were: invite_hierarchy_collection_handler and invite_user_collection_handler, where the first handle the hierarchy calls (INSTITUTION_PARENT and INSTITUTION_CHILDREN) and the second handler (USER and USER_ADM).

**TODO/FIXME:** n/a

### CODE REVIEW

File: backend/handlers/invite_hierarchy_collection_handler.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+"""."""
+
+import re
```
Comment (mayzabeel) [inline backend/handlers/invite_hierarchy_collection_handler.py]: update libraries table

File: backend/handlers/invite_hierarchy_collection_handler.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from models import InviteFactory
+from service_entities import enqueue_task
+
+__all__ = ['InviteHierachyCollectionHandler']
+
+class InviteHierachyCollectionHandler(BaseHandler):
+    
+    @login_required
+    @json_response
+    def post(self, user):
```
Comment (mayzabeel) [inline backend/handlers/invite_hierarchy_collection_handler.py]: insert documentation

File: backend/handlers/invite_user_collection_handler.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+from models import InviteFactory
+from service_entities import enqueue_task
+
+__all__ = ['InviteUserCollectionHandler']
+
+class InviteUserCollectionHandler(BaseHandler):
+    
+    @login_required
+    @json_response
+    def post(self, user):
```
Comment (mayzabeel) [inline backend/handlers/invite_user_collection_handler.py]: insert documentation

File: backend/handlers/invite_user_collection_handler.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        institution = ndb.Key(urlsafe=invite['institution_key']).get()
+        can_invite_members = user.has_permission(
+            "invite_members", institution.key.urlsafe())
+
+        Utils._assert(not can_invite_members,
+                        "User is not allowed to send invites", NotAuthorizedException)
+
+        invites = []
+        @ndb.transactional(xg=True, retries=10)
+        def process_invites(emails, invite, current_institution_key):
```
Comment (mayzabeel) [inline backend/handlers/invite_user_collection_handler.py]: insert documentation

File: backend/handlers/invite_user_collection_handler.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        self.response.write(json.dumps(
+            {'msg': 'The invites are being processed.', 'invites' : invites}))
+
+
+def createInvite(data):
+    """Create an invite."""
+    invite = InviteFactory.create(data, data['type_of_invite'])
+    invite.put()
+
+    if(invite.stub_institution_key):
```
Comment (mayzabeel) [inline backend/handlers/invite_user_collection_handler.py]: this kind of invite never has a stub institution or it is possible?

File: backend/handlers/invite_user_collection_handler.py (hunk 3 above)
Comment (Luiz-FS) [reply backend/handlers/invite_user_collection_handler.py]: Done.

File: backend/handlers/invite_hierarchy_collection_handler.py (hunk 2 above)
Comment (Luiz-FS) [reply backend/handlers/invite_hierarchy_collection_handler.py]: Done.

File: backend/handlers/invite_user_collection_handler.py (hunk 4 above)
Comment (Luiz-FS) [reply backend/handlers/invite_user_collection_handler.py]: Done.

File: backend/handlers/__init__.py [hunk 6]
```diff
@@ -38,6 +38,8 @@
 from .user_request_collection_handler import *
 from .user_timeline_handler import *
 from .vote_handler import *
+from .invite_hierarchy_collection_handler import *
```
[the remark this answers is not in the document]
Comment (Luiz-FS) [reply backend/handlers/__init__.py]: Done

File: backend/handlers/invite_hierarchy_collection_handler.py [hunk 7]
```diff
@@ -11,13 +12,18 @@
...
 
-class InviteHierachyCollectionHandler(BaseHandler):
+class InviteHierarchyCollectionHandler(BaseHandler):
     
     @login_required
     @json_response
     def post(self, user):
+        """Handler POST invites.
+        
+        This method creates contives for:
```
Comment (mayzabeel) [inline backend/handlers/invite_hierarchy_collection_handler.py]: create invites

File: backend/handlers/invite_user_collection_handler.py [hunk 8]
```diff
@@ -18,6 +19,12 @@ class InviteUserCollectionHandler(BaseHandler):
     @login_required
     @json_response
     def post(self, user):
+        """Handle POST invites.
+        
+        This method creates contives for:
```
Comment (mayzabeel) [inline backend/handlers/invite_user_collection_handler.py]: invites

**Pull request merged**